stripe/_stripe_client.py

```diff
@@ -280,3 +284,62 @@ def construct_event(
         )
 
         return event
+
+    def raw_request(self, method_: str, url_: str, **params):
+        params = params.copy()
+        options, params = extract_options_from_dict(params)
+        api_mode = params.pop("api_mode", None)
+        base_address = params.pop("base", "api")
+
+        stripe_context = params.pop("stripe_context", None)
+
+        # stripe-context goes *here* and not in api_requestor. Properties
+        # go on api_requestor when you want them to persist onto requests
+        # made when you call instance methods on APIResources that come from
+        # the first request. No need for that here, as we aren't deserializing APIResources
+        if stripe_context is not None:
+            options["headers"] = options.get("headers", {})
+            assert isinstance(options["headers"], dict)
+            options["headers"].update({"Stripe-Context": stripe_context})
+
+        rbody, rcode, rheaders = self._requestor.request_raw(
+            method_,
+            url_,
+            params=params,
+            options=options,
+            base_address=base_address,
+            api_mode=api_mode,
+            usage=["raw_request"],
+        )
+
+        return self._requestor._interpret_response(rbody, rcode, rheaders)
+
+    async def raw_request_async(self, method_: str, url_: str, **params):
+        params = params.copy()
+        options, params = extract_options_from_dict(params)
+        api_mode = params.pop("api_mode", None)
+        base_address = params.pop("base", "api")
+
+        rbody, rcode, rheaders = await self._requestor.request_raw_async(
+            method_,
+            url_,
+            params=params,
+            options=options,
+            base_address=base_address,
+            api_mode=api_mode,
+            usage=["raw_request"],
+        )
+
+        return self._requestor._interpret_response(rbody, rcode, rheaders)
+
+    def deserialize(
+        self,
+        resp: Union[StripeResponse, Dict[str, Any]],
+        params: Optional[Dict[str, Any]] = None,
+    ) -> StripeObject:
+        return _convert_to_stripe_object(
+            resp=resp,
+            params=params,
+            requestor=self._requestor,
+            api_mode="V1",
```

Comment:
This needs to be compatible with v2. What do you think about passing this as a param?

Reply:
resolved in https://github.com/stripe/stripe-python/pull/1399